### DISCUSSION TITLE
Invisible box

### DISCUSSION THREAD

**Discussion by indigoviolet** [jsvine/pdfplumber#996] [Ask for help with specific PDFs]
How can I identify that the inner rect below is invisible? There are other rects in this PDF with exactly the same `fill, stroke, stroking_color, linewidth and non_stroking_color` that are visible. But this inner rect is not. 

```python
page = pdf.pages[0]
page = page.crop((550, 170, 585, 200))
im = page.to_image(resolution=300)
for e in page.rects:
    print(e)
    _ = im.draw_rect(e, stroke="red", stroke_width=1)


im
im.reset()
```

![image](https://github.com/jsvine/pdfplumber/assets/114010/c903078e-dc23-4a6d-bb04-b3eaec91ec4b)
![image](https://github.com/jsvine/pdfplumber/assets/114010/dca6f0fd-112d-46c7-916b-9a5e7191b92d)

```
{'x0': 562.001, 'y0': 603.498, 'x1': 570.001, 'y1': 611.498, 'width': 8.0, 'height': 8.0, 'pts': [(562.001, 188.50199999999995), (570.001, 188.50199999999995), (570.001, 180.50199999999995), (562.001, 180.50199999999995)], 
'linewidth': 0.5, 'stroke': True, 'fill': False, 'evenodd': False, 'stroking_color': (0, 0, 0, 1), 'non_stroking_color': (0, 0, 0, 1), 'object_type': 'rect', 'page_number': 1, 'stroking_pattern': None, 'non_stroking_pattern': None, 
'top': 180.50199999999995, 'bottom': 188.50199999999995, 'doctop': 180.50199999999995}


{'x0': 557.89, 'y0': 599.24, 'x1': 575.89, 'y1': 617.24, 'width': 18.0, 'height': 18.0, 'pts': [(557.89, 192.76), (575.89, 192.76), (575.89, 174.76), (557.89, 174.76)], 
'linewidth': 0, 'stroke': False, 'fill': True, 'evenodd': False, 'stroking_color': None, 'non_stroking_color': (1,), 'object_type': 'rect', 'page_number': 1, 'stroking_pattern': None, 'non_stroking_pattern': None, 
'top': 174.76, 'bottom': 192.76, 'doctop': 174.76}

```

**Comment by indigoviolet** [comment]
[test.pdf](https://github.com/jsvine/pdfplumber/files/12698108/test.pdf)

Here's the PDF if that's helpful.

**Reply by samkit-jain** [reply]
Hi @indigoviolet Appreciate your interest in the library and thanks for sharing the PDF. Looking at the properties you've shared, seems like a lot of properties are different.
 
|                      | Top dict     | Bottom dict |
|----------------------|--------------|-------------|
| `width`              | 8.0          | 18.0        |
| `height`             | 8.0          | 18.0        |
| `linewidth`          | 0.5          | 0           |
| `stroke`             | True         | False       |
| `fill`               | False        | True        |
| `stroking_color`     | (0, 0, 0, 1) | None        |
| `non_stroking_color` | (0, 0, 0, 1) | (1,)        |

You can use either of these to differentiate.
